homeassistant/components/sonos/media_player.py

```diff
@@ -337,8 +337,16 @@ def __init__(self, player):
     async def async_added_to_hass(self):
         """Subscribe sonos events."""
         await self.async_seen()
+
         self.hass.data[DATA_SONOS].entities.append(self)
 
+        def _rebuild_groups():
+            """Build the current group topology."""
+            for entity in self.hass.data[DATA_SONOS].entities:
+                entity.update_groups()
+
+        self.hass.async_add_job(_rebuild_groups)
```

Comment:
Please use `self.hass.async_add_executor_job` instead of `self.hass.async_add_job`.